metricbeat/module/kubernetes/state_container/state_container.go

```diff
@@ -132,9 +130,26 @@ func (m *MetricSet) Fetch() ([]common.MapStr, error) {
 				event["cpu.limit.nanocores"] = limitCores * nanocores
 			}
 		}
-	}
 
-	return events, err
+		var moduleFieldsMapStr common.MapStr
+		moduleFields, ok := event[mb.ModuleDataKey]
```

Comment:
@odacremolbap I wonder if with your refactoring we will be able to remove this "trick".